### ISSUE TITLE
Crash when point picking on a mesh

### ISSUE THREAD

**Issue by benoitrosa** [CloudCompare/CloudCompare#713]
Hi,

When I try to use the point picking tool, CloudCompare crashes with a core dump. 

Steps : 
- Load a mesh in CloudCompare
- Select it and make vertices visible
- Choose the point picking tool
- Click anywhere on the mesh
- The program crashes

Before the crash, a dialog appears asking whether I want to build an octree to speed up the process. The crash appears independently of what I select. The message in the CLI is : 

> CloudCompare: [I removed home dir]/CloudCompare/qCC/ccPointPickingGenericInterface.cpp:126: virtual void ccPointPickingGenericInterface::onItemPicked(const ccPickingListener::PickedItem&): Assertion `false' failed.
> zsh: abort (core dumped)  CloudCompare

I am on ubuntu 16.04, with gcc 5.4.0/ CloudCompare was compiled from source. I tried the latest commit from gitHub, as well as the commit tagged v2.9.1, with the exact same results. Glxinfo gives the following for my graphic renderer : OpenGL renderer string: Mesa DRI Intel(R) HD Graphics 530 (Skylake GT2). 

It's also probably worth noting that the "Trace a polyline by point picking" tool works as expected without causing a core dump.

**Comment by benoitrosa** [comment]
A small update : the crash does not appear with the 2.8.0 release, compiled from source.

**Comment by rjanvier** [comment]
Can't reproduce it on 2.10 alpha (the current snap package).

**Comment by rjanvier** [comment]
ok from code it seems that point picking for mesh is not implemented 
`        else if (pi.entity->isKindOf(CC_TYPES::MESH))
        {
                //NOT HANDLED: 'POINT_PICKING' mode only for now
                assert(false);
        }
`

**Comment by benoitrosa** [comment]
That code is the same in the 2.8.1 branch, but the point picking tool on the same mesh is working. 

Visually, when I start the point picking in the 2.8.1 version, it shows me a point cloud with all the vertices in the selected mesh, from which I can select. In the latest version (master branch), this point cloud does not show. Maybe the bug is in a preprocessing step which was removed ?

**Comment by rjanvier** [comment]
![capture d ecran de 2018-04-20 10-47-30](https://user-images.githubusercontent.com/1537100/39040907-43cafc3e-4488-11e8-8040-0140a9718b62.png)

Points are ok here

**Comment by benoitrosa** [comment]
Interesting : it does indeed run well when using the snap-installed package (2.10.0 alpha).

Perhaps it has to do with my CMake configuration ? It would be interesting to see whether other people can reproduce this crash when building from source using the master branch. 

Note: I said before that the bug does not appear when using the 2.8.1 version, but actually it works only once. It reproduces the same crash f I try to pick a second point on the same mesh.

**Comment by rjanvier** [comment]
I build it from source and it works too

**Comment by rjanvier** [comment]
to summarize: It will crash if you compile in debug mode because it will assert. In release it will fail silently. To be more explicit, by default CC is compiled in Debug mode, so yes try to use -DCMAKE_BUILD_TYPE=Release and it will "work".

if you can't see points of point cloud, try to increase the size of the points.

**Comment by benoitrosa** [comment]
Looking back at my CMake outputs, I noticed a warning considering LibQT5Svg. 

After installing libqt55svg-dev and recompiling, the program works as expected. 

I'm not familiar with the build structure of CloudCompare, but if this lib is required for having a working software, perhaps its absence should trigger more than a warning ?

**Comment by rjanvier** [comment]
There is two things here: 
- First you compiled CC in debug mode, and had and error that was perfectly normal. it is solved by the fact you now compile in release mode, not because you add another lib. 
- For libqt55svg, yes maybe you are right, it should be mandatory.

**Comment by benoitrosa** [comment]
Actually, the error and crash disappeared when I compiled after installing libqt5svg-dev, but I was still compiling in debug mode. I just recompiled in release now, and of course it still works. 

Anyway, thanks for the help !

**Comment by rjanvier** [comment]
I recompiled without libqt svg and obviously it has nothing to deal with your error.

**Comment by dgirardeau** [comment]
Indeed, Qt SVG component was only added recently because of 2 SVG icons added by Andy :D

It is not 100% mandatory (and I thought I updated the CMake script for Qt integration but I may have forgotten to push that) and it clearly has nothing to do with this error.

**Comment by dgirardeau** [comment]
But thanks for the feedback anyway. It's always important to report (potential) bugs ;)

**Comment by lucivpav** [comment]
Is point picking on a mesh supported now? I cannot get it to work. The "Build octree" dialog also does not show, even if it is set to appear each time.

**Comment by asmaloney** [comment]
Hey @lucivpav - can you please open a new issue and fill out the details so we have more information?

Thank you.

**Comment by lucivpav** [comment]
Hi Andy, I just found out that I had to check 'Vertices' in the DB Tree pane, in order for the point picking information to show up.

Cheers.

**Comment by dgirardeau** [comment]
Well, normally with the recent versions of CC (2.10 and 2.11) you can directly pick point on meshes (you don't need to pick the vertices).
